lib/trace-agent.js

```diff
@@ -148,6 +148,44 @@ TraceAgent.prototype.runInSpan = function(name, labels, fn) {
 };
 
 /**
+ * Run the provided function in a new root span with the provided
+ * name. As with runInSpan, if the provided function accepts a parameter,
+ * it is assumed to be asynchronous, and a callback will be passed
+ * to the function to terminate the root span.
+ * @param {string} name The name of the resulting root span.
+ * @param {Object<string, string}>=} labels Labels to be attached
+ *   to the resulting span.
+ * @param {function(function()=)} fn The function to trace.
+ */
+TraceAgent.prototype.runInRootSpan = function(name, labels, fn) {
+  if (typeof(labels) === 'function') {
+    fn = labels;
+    labels = undefined;
+  }
+  this.namespace.run((function () {
+    if (cls.getRootContext()) {
+      this.logger.error('Can\'t nest root spans');
+      return;
+    }
+    var span = this.createRootSpanData(name, null, null, 0, 'SPAN_KIND_UNSPECIFIED');
+    if (labels) {
+      Object.keys(labels).forEach(function(key) {
+        span.addLabel(key, labels[key]);
+      });
+    }
+    if (fn.length === 0) {
+      fn();
+      span.close();
+    } else {
+      fn(function () {
+        span.close();
+      });
+      cls.setRootContext(null);
```

Comment:
Originally it was to help with the check to make sure root spans couldn't be nested, but it seems like this is unnecessary. I'll remove it.